### ISSUE TITLE
Error with Nuxt + cookie-storage : document is not defined

### ISSUE THREAD

**Issue by RomainLK** [feathers-plus/feathers-vuex#97]
I'm adding feathers-vuex to my Nuxt app and I'm encountering an issue with the example using cookie storage. Basically, I get:

```document is not defined```

line 75 of node_modules/cookie-storage/lib/cookie-storage.js

This error seems normal, as just like `window`, `document` shouldn't be present server side.
But following #50, it also seems that some people got it to work using cookie-storage, so I'm left quite puzzled. Common sense tells me that the cookie-storage solution shouldn't work.

Nuxt version: 1.3.0
Feathers-vuex: 1.1.4
Cookie-storage: 3.1.0
Feathers: 3.1.3
Feathers-rest-client: 1.3.3
authentication-client:1.0.2

feathers-client.js:
```
import feathers from '@feathersjs/feathers'
import rest from '@feathersjs/rest-client'
import auth from '@feathersjs/authentication-client'
import axios from 'axios'
import { CookieStorage } from 'cookie-storage'
const restClient = rest('http://localhost:3030')
const feathersClient = feathers().configure(restClient.axios(axios))
feathersClient.configure(auth({ storage: new CookieStorage() }))

export default feathersClient
```

The client is used in an asyncData function of a Nuxt page, throught the store with the 'auth/authenticate' action.

Note that it does work when outside of asyncData, but that would be because we are on the browser.

Am I missing something, or perhaps should I be searching for some kind of universal cookies solution?

**Comment by RomainLK** [comment]
Ok, actually I kind of have an idea of the problem after reading the code of initAuth which already handles cookies server side.

I assume that it's by design that it's not possible to call 'auth/authenticate' on the server. If so, then you can close this issue.

**Comment by RomainLK** [comment]
Ok, I have my solution for the following case:

1. Log on the browser to my Feathers API and store JWT in cookie. Feathers' client use cookie-storage
2. Access to page with asyncData. The async data block will fetch via REST data from a Feather server while being authenticated with the JWT from the cookie.

Architecture wise, my Nuxt application is not a Feather application, as I'm currently adding a Feather client to it; I think that those who will use Nuxt as a Feather/Express middleware may have less issues since feather would be able to parse the cookie by itself.

Now, for storage, I just used a mock of cookie-storage.

The trick is during nuxtServerInit:

```javascript
  nuxtServerInit ({ commit, dispatch, state }, { req }) {
    return initAuth({
      commit,
      dispatch,
      req,
      moduleName: 'auth',
      cookieName: 'feathers-jwt'
    }).then(result => {
      feathersClient.set('accessToken', state.auth.accessToken)
      return result
    })
  }
```

Usually feathersClient will fetch the JWT from storage, but since it's already recovered by feathers-vuex, I just added it to the featherClient.

This is a bit hacky, and I'm still thinking of security concern, like shared feathersClient...
